cool-core/src/main/java/com/nus/cool/core/cohort/funnel/FunnelQuery.java

```diff
@@ -21,40 +21,36 @@ public List<BirthSequence> getStages() {
 		return stages;
 	}
 
-	public void setStages(List<BirthSequence> stages) {
-		this.stages = stages;
-	}
+	// public void setStages(List<BirthSequence> stages) {
+	//	 this.stages = stages;
+	// }
 
 	public boolean isOrdered() {
 		return isOrdered;
 	}
 
-	public void setOrdered(boolean isOrdered) {
-		this.isOrdered = isOrdered;
-	}
+	// public void setOrdered(boolean isOrdered) {
+	// 	 this.isOrdered = isOrdered;
+	// }
 
 	public String getDataSource() {
 		return dataSource;
 	}
 
-	public void setDataSource(String dataSource) {
-		this.dataSource = dataSource;
-	}
+	// public void setDataSource(String dataSource) {
+	//   this.dataSource = dataSource;
+	// }
 
 	public String getInputCohort() {
 		return inputCohort;
 	}
 
-	public void setInputCohort(String inputCohort) {
-		this.inputCohort = inputCohort;
-	}
+	// public void setInputCohort(String inputCohort) {
+	//	 this.inputCohort = inputCohort;
+	// }
 
 	@JsonIgnore
-	public boolean isValid() throws IOException {
-		if ((stages != null) && (dataSource != null)){
-			return true;
-		} else{
-			throw new IOException("[x] Invalid funnel query.");
-		}
+	public boolean isValid() {
```

Comment:
Please revert this change, i.e., keep the original isValid() function with IOException.